profiling/paper_timing_tables/timing_comparison.py

```diff
@@ -0,0 +1,437 @@
+    make_fake_TOA2 = "zima --startMJD 53478 --duration 700 --freq 1400 2000 --ntoa 1000 NGC6440E.par NGC6440E_fake1k.tim"
+    make_fake_TOA3 = "zima --startMJD 53478 --duration 700 --freq 1400 2000 --ntoa 10000 NGC6440E.par NGC6440E_fake10k.tim"
+    make_fake_TOA4 = "zima --startMJD 53478 --duration 700 --freq 1400 2000 --ntoa 100000 NGC6440E.par NGC6440E_fake100k.tim"
+    # call operations on command line
+    print("Making fake TOAs...")
+    subprocess.call(
+        make_fake_TOA1, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL
+    )
+    subprocess.call(
+        make_fake_TOA2, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL
+    )
+    subprocess.call(
+        make_fake_TOA3, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL
+    )
+    subprocess.call(
+        make_fake_TOA4, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL
+    )
+
+    ptimes_nopickle = []
+    ptimes_pickle = []
+    ttimes = []
+    t2times = []
+
+    for tim in timfiles:
+        print("Running PINT fitting w/o pickling...")
+        # run PINT w/o pickling and average time over 5 runs
+        pintrun("NGC6440E.par", tim, ptimes_nopickle, pickle=False, fitter="wls")
+
+        print("Running PINT w/ pickling...")
+        # run PINT with pickling and average time over 5 runs
+        subprocess.call(
+            "pintempo --usepickle NGC6440E.par " + tim,
+            shell=True,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )  # create pickle file
+        pintrun("NGC6440E.par", tim, ptimes_pickle, pickle=True, fitter="wls")
+        print("Running TEMPO...")
+        temporun("NGC6440E.par", tim, ttimes, fitter="wls")
+        print("Running Tempo2...")
+        tempo2run("NGC6440E.par", tim, t2times)
+
+    # create table 7 in PINT paper
+    ntoas = [100, 1000, 10000, 100000]
+    simple_comparison = Table(
+        (ntoas, ttimes, t2times, ptimes_nopickle, ptimes_pickle),
+        names=(
+            "Number of TOAs",
+            "TEMPO (sec)",
+            "Tempo2 (sec)",
+            "PINT - No Pickle (sec)",
+            "PINT - Pickle (sec)",
+        ),
+    )
+    ascii.write(
+        simple_comparison,
+        "simple_tables.pdf",
+        Writer=ascii.Latex,
+        latexdict={"tabletype": "table*"},
+        overwrite=True,
+    )
+
+    # time the individual PINT functions
+    importtimes = []
+    getTOAs_nopickle = []
+    getTOAs_pickle = []
+    fittimes = []
+
+    # setup
+    m = models.get_model("NGC6440E.par")
+    use_planets = False
+    if m.PLANET_SHAPIRO.value:
+        use_planets = True
+    model_ephem = "DE421"
+    if m.EPHEM is not None:
+        model_ephem = m.EPHEM.value
+
+    # time import statements in pintempo script
+    print("timing imports...")
+    total = 0
+    for i in range(MAXIT):
```

Comment:
This will not work. Python imports are aggressively cached, so these are basically no-ops.  To measure the import speed, you need to make a script with just those import statements and run it as a separate process.